HeterogeneousCore/CUDACore/interface/ScopedContextBase.h

```diff
@@ -0,0 +1,62 @@
+#ifndef HeterogeneousCore_CUDACore_ScopedContextBase_h
+#define HeterogeneousCore_CUDACore_ScopedContextBase_h
+
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/device_unique_ptr.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/SharedStreamPtr.h"
+
+namespace cms {
+  namespace cuda {
+    class ProductBase;
+
+    class ScopedContextBase {
```

Comment:
If this (base) class gets exposed to users, I'm tempted to rename it to just `ScopedContext`. (Even if I would then have to figure out what to do for current `ScopedContext.h`. One option is to merge the headers back together and make it C++14-compatible).